src/lambda_streams/Numbers.java

```diff
@@ -9,83 +11,281 @@
+            double sum = 0;
+            for (int n = 0; n< Numbers.nums.size(); n++)
+                sum += Numbers.nums.get(n);
+            return sum;
+        };
+        System.out.println(val4.calc(nums));
+
+        LambdaCalc<List<Integer>, Double> val5 = (List<Integer> i) -> {
+            double sum = nums.get(0);
+            for (int n = 1; n<nums.size(); n++)
+                sum -= nums.get(n);
+            return sum;
+        };
+        System.out.println(val5.calc(nums));
+
+        LambdaCalc<List<Integer>, Long> val6 = (List<Integer> i) -> {
+            long product = nums.get(0);
+            for (int n = 1; n<nums.size(); n++)
+                product *= nums.get(n);
+            return product;
+        };
+        System.out.println(val6.calc(nums));
+
+        LambdaCalc<List<Integer>, Double> val7 = (List<Integer> i) -> {
+            double product = nums.get(0);
+            for (int n = 1; n<nums.size(); n++)
+                product /= nums.get(n);
+            return product;
+        };
+        System.out.println(val7.calc(nums));
+
+        LambdaCalc<List<Integer>, Integer> val8 = (List<Integer> i) -> {
+            List<Integer> sortedList = new ArrayList<>(nums);
+            Collections.sort(sortedList);
+            return sortedList.get(sortedList.size()-1);
+        };
+        System.out.println(val8.calc(nums));
+
+        LambdaCalc<List<Integer>, Integer> val9 = (List<Integer> i) -> {
+            List<Integer> sortedList = new ArrayList<>(nums);
+            Collections.sort(sortedList);
+            return sortedList.get(0);
+        };
+        System.out.println(val9.calc(nums));
+
+        LambdaCalc<List<Integer>, Integer> val10 = (List<Integer> a) -> {
+            int i = 6;
+            int j = 3;
+            System.out.println("Nums array list: " + nums);
+            if (nums.get(i) > nums.get(j)){
+                System.out.println(nums.get(i) + " is > " + nums.get(j));
+                return 1;
+            } else if (nums.get(i) == nums.get(j)) {
+                System.out.println(nums.get(i) + " = " + nums.get(j));
+                return 0;
+            } else {
+                System.out.println(nums.get(i) + " is < " + nums.get(j));
+                return -1;
+            }
+        };
+        System.out.println(val10.calc(nums));
+
+        LambdaOneArg<Integer> val11 = (a) -> {
+            a = 17;
+            System.out.println("Nums length is " + nums.size());
+            List<Integer> newAddedNums = new ArrayList<Integer>(nums);
+            newAddedNums.add(a);
+            System.out.println("Added nums length is " + newAddedNums.size());
+            return a;
+        };
+        System.out.println("New element is " + val11.oneArg(0));
+
+
     }
 
-    static boolean isOdd(int i) {
+    static <i> boolean isOdd(List<Integer> i) {
         //determine if the value at the index i is odd.  return true if yes, return false if  no.
```

Comment:
The intent was to accept an 'index' argument, then determine if the value at that location in the array is odd.  Seems the method instead identifies the first odd value.  This approach made the exercise a bit more challenge, so KUDOs for pulling this off.  =0)